src/util/get-body-size.ts

```diff
@@ -0,0 +1,42 @@
+import type { Readable } from 'node:stream';
+import { HttpBody } from '../types';
+import is from '@sindresorhus/is';
+
+export function getBodySize(
+  body: HttpBody | undefined,
+  headers: { [name: string]: string } | undefined,
+): number | undefined {
+  function isFormData(body: unknown): body is FormData {
+    return is.nodeStream(body) && is.function_((body as FormData).getBoundary);
+  }
```

Comment:
this looks specifically for [this `FormData`](https://github.com/form-data/form-data), not the [web native](https://developer.mozilla.org/en-US/docs/Web/API/FormData) or [node native](https://nodejs.org/dist/latest-v20.x/docs/api/globals.html#class-formdata) versions.

we use `FormData` in jsforce.